pontoon/base/static/js/translate.js

```diff
@@ -107,7 +107,7 @@ var Pontoon = (function (my) {
                       (this.approved ? this.approved_user ?
                         'Approved by ' + this.approved_user : '' : 'Approve') +
                       '"></button>' +
-                      ((self.user.email && (self.user.email === this.email)) ? '<button class="delete fa" title="Delete"></button>' : '') +
+                      ((self.user.email && (self.user.email === this.email) || self.user.isLocaleManager) ? '<button class="delete fa" title="Delete"></button>' : '') +
```

Comment:
We should actually allow all translators to remove translations, so replace `isLocaleManager` with `isTranslator`.